src/scry/client.cr

```diff
@@ -1,10 +1,20 @@
+require "./request"
+
 module Scry
   class Client
     alias ClientMessage = Protocol::Initialize | Protocol::ResponseMessage | Protocol::NotificationMessage
 
-    getter io
+    getter io_out
+
+    def initialize(@io_in : IO, @io_out : IO)
+    end
+
+    def self.new
```

Comment:
I'm not sure about this default constructor, I think it's better to explicitly initialize the `Client` instead of using defaults (which appear to be right for their current use case, but can be misleading for others)